analytics/src/main/java/com/segment/analytics/Analytics.java

```diff
@@ -284,14 +285,25 @@ public void run() {
 
     logger.debug("Created analytics client for project with tag:%s.", tag);
 
+    PackageInfo packageInfo = getPackageInfo(application);
+    final String currentVersion = packageInfo.versionName;
+    final int currentBuild = packageInfo.versionCode;
+
     activityLifecycleCallback =
         new Application.ActivityLifecycleCallbacks() {
           final AtomicBoolean trackedApplicationLifecycleEvents = new AtomicBoolean(false);
+          final AtomicInteger numberOfActivities = new AtomicInteger(1);
+          final AtomicBoolean isChangingActivityConfigurations = new AtomicBoolean(false);
+          final AtomicBoolean firstLaunch = new AtomicBoolean(false);
 
           @Override
           public void onActivityCreated(Activity activity, Bundle savedInstanceState) {
+            runOnMainThread(IntegrationOperation.onActivityCreated(activity, savedInstanceState));
+
             if (!trackedApplicationLifecycleEvents.getAndSet(true)
                 && shouldTrackApplicationLifecycleEvents) {
+              numberOfActivities.set(0);
+              firstLaunch.set(true);
```

Comment:
These reset the atomic variables for native Android apps. `onActivityCreated` is not called when this library is used by the React Native wrapper.